rotkehlchen/api/rest.py

```diff
@@ -4547,11 +4548,13 @@ def _get_exchange_staking_or_savings_history(
                 query_filter=table_filter,
             )
             value_query_filters, value_bindings = value_filter.prepare(with_pagination=False, with_order=False)  # noqa: E501
-            usd_value, amounts = history_events_db.get_value_stats(
+            asset_amounts_and_value = history_events_db.get_amount_and_value_stats(
                 cursor=cursor,
                 query_filters=value_query_filters,
                 bindings=value_bindings,
+                counterparty=CPT_KRAKEN,
             )
+            usd_value = sum(x[2] for x in asset_amounts_and_value)
```

Comment:
you can just return the total usd value from the method call. that way you don't need to loop twice.